contracts/lifecycle/TimeFrame.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Provides generic functionality for a TimeFrame.
+ * @author Sam Ruberti <sam@GRAFFT.io>
+ * @dev This library provides useful methods for governing a start and an end time.
+ *
+ * WARNING: This should NOT be used for contracts that depend on time for an outcome where a miner can benefit.
+ * One example would be a lottery where the first submission of an answer to a problem is rewarded. In this case the
+ * contract would be vulnerable to malicious miners who could read the answer of a not-yet-mined transaction then
+ * substitute their own answer in a new transaction before the victim's transaction, stealing the reward in the process.
+ * The miner could steal the reward.
+ *
+ * SAFE USE: TimeFrame CAN be used for approximate times that govern periods when methods and actions may occur.
+ * An example is an escrow that contains a start and expiration date. You can read about timestamp security
+ * vulnerabilities here:
+ * github.com/ethereumbook/ethereumbook/blob/develop/09smart-contracts-security.asciidoc#block-timestamp-manipulation
+ */
+
+
+/* solhint-disable not-rely-on-time, security/no-block-members */
+library TimeFrame {
+    struct Epoch {
+        uint256 start;
+        uint256 end;
+    }
+
+    function hasStarted(Epoch storage epoch) internal view returns (bool) {
+        return now >= epoch.start;
+    }
+
+    function isActive(Epoch storage epoch) internal view returns (bool) {
+        return now >= epoch.start && now <= epoch.end;
+    }
+
+    function hasEnded(Epoch storage epoch) internal view returns (bool) {
+        return now > epoch.end;
+    }
+
+    function timeUntilStart(Epoch storage epoch) internal view returns (uint256) {
+        require(now <= epoch.start, "Already started");
```

Comment:
```suggestion
        require(now < epoch.start, "Already started");
```

Nitpicking, but `hasStarted` returns true when `now == start`

Reply:
is this concept applies to all of the require statements or just timeUntilStart?
for now, I just fixed this function

Reply:
I think this is the only one, they should all be consistent: if `hasStarted` returns true when `now == start`, then `timeUntilStart` should revert under that same condition.